projects/lights_control/test/test_lights_signal.c

```diff
@@ -0,0 +1,122 @@
+#include "test_helpers.h"
+#include "unity.h"
+
+#include "lights_blinker.h"
+#include "lights_events.h"
+#include "lights_signal_fsm.h"
+
+#define TEST_LIGHTS_SIGNAL_FSM_DURATION 300
+
+typedef enum {
+  TEST_LIGHTS_SIGNAL_FSM_CMD_OFF = 0,
+  TEST_LIGHTS_SIGNAL_FSM_CMD_ON,
+  NUM_TEST_LIGHTS_SIGNAL_FSM_CMDS
+} TestLightsSignalFsmCmd;
+
+static LightsBlinkerDuration s_duration_ms = TEST_LIGHTS_SIGNAL_FSM_DURATION;
+
+static LightsSignalFsm s_signal_fsm = { 0 };
+
+void setup_test(void) {
+  event_queue_init();
+  lights_signal_fsm_init(&s_signal_fsm, s_duration_ms);
+}
+
+static void prv_assert_raised_event(LightsEvent event, LightsEventGpioPeripheral peripheral) {
```

Comment:
I'd recommend making this a macro or adding extra logging so you can figure out where in the test the unexpected behavior occurred.